### ISSUE TITLE
Correct CMake debug for global-workflow build system

### ISSUE THREAD

**Issue by HenryRWinterbottom** [NOAA-EMC/GSI-Monitor#118]
The `gsi_monitor` CMake file does not support `debug` build types. As noted in NCO [Bugzilla #1218](http://www2.spa.ncep.noaa.gov/bugzilla/show_bug.cgi?id=1218), all build systems must include `debug` and `clean` options for the respective packages .

To resolve this issue, a `debug` options should be added using the CMake option `CMAKE_BUILD_TYPE`. Currently, it defaults to `Release`.

**Comment by CatherineThomas-NOAA** [comment]
@EdwardSafford-NOAA - Has this issue been addressed?

**Comment by EdwardSafford-NOAA** [comment]
@CatherineThomas-NOAA gosh no it has not.  I'm on leave 10/6-10/21, will it be too late to address it when I'm back?

**Comment by CatherineThomas-NOAA** [comment]
@EdwardSafford-NOAA - These aren't changes that would affect the science of GFSv17 retrospectives, so it can wait until after your leave.  Enjoy!

**Comment by EdwardSafford-NOAA** [comment]
Upon review I'm not exactly sure what changes are required.  

The repo's top level CMakeLists.txt file does support building in `Debug`.  The user build script (`ush/build.sh`) only builds in `Release` but manually changing that type to `Debug` does work, albeit with bundles of warning messages.  Is the issue that this script needs to allow selection of `Debug` build type using a parameter and/or that all the debug warnings need to be corrected?  Or is it something else that I'm missing?

@HenryRWinterbottom @CatherineThomas-NOAA

**Comment by CatherineThomas-NOAA** [comment]
Digging through old PRs, it looks like this particular issue was addressed PR #119 and can likely be closed.

That said,  NCO will require the system to run end-to-end in debug mode and the verfrad (radmon) step is currently failing in debug mode (https://github.com/NOAA-EMC/GSI-Monitor/issues/197).  So that is a separate piece that will need to be addressed.

**Comment by EdwardSafford-NOAA** [comment]
@CatherineThomas-NOAA Thanks for the update.  I'm going to keep this issue open -- I've started removing the warnings from debug mode and I can do that work under this issue.  But I'll make [#197](https://github.com/NOAA-EMC/GSI-Monitor/issues/197) the higher priority and get to work on that next thing.

**Comment by CatherineThomas-NOAA** [comment]
@EdwardSafford-NOAA - I believe this issue can be closed.  Is that correct?

**Comment by EdwardSafford-NOAA** [comment]
@CatherineThomas-NOAA Yes, this issue can be closed -- debug is supported by CMake now.
